aemcmc/nuts.py

```diff
@@ -0,0 +1,166 @@
+from collections import defaultdict
+from typing import Dict, List
+
+import aehmc
+import aesara
+import aesara.tensor as at
+import numpy as np
+from aehmc.utils import RaveledParamsMap
+from aeppl import factorized_joint_logprob
+from aeppl.transforms import TransformValueOpt, _default_transformed_rv
+from aesara.tensor.random import RandomStream, RandomVariable
+from aesara.tensor.var import TensorVariable
+
+from aemcmc.utils import ModelInfo
+
+
+def get_transform(rv: RandomVariable):
+    """Get the default transform associated with the random variable.
+
+    Algorithms in the HMC family perform better when the support of every
+    random variable in the model is the real line.
+    """
+    trans = _default_transformed_rv(rv.owner.op, rv.owner)
+    if trans:
+        return trans.op.transform
+    else:
+        return None
+
+
+def transform_var(rv: RandomVariable, vv: TensorVariable, forward=True):
+    """Transform variables back and forth from and to the transformed space.
+
+    Parameters
+    ----------
+    rv
+        The corresponding random variable.
+    var
+        The current realization of the randomvariable
+    forward:
+        If `true` transforms the value to the transformed space, back from
+        the transformed space otherwise.
+
+    """
+    trans = _default_transformed_rv(rv.owner.op, rv.owner)
+    if trans:
+        if forward:
+            res = trans.forward(vv, *rv.owner.inputs)
+        else:
+            res = trans.backward(vv, *rv.owner.inputs)
+        if vv.name:
+            res.name = f"{vv.name}_trans"
+        return res
+    else:
+        return vv
+
+
+def nuts(
+    rng: RandomStream,
+    model: ModelInfo,
+    sampled_rvs: Tuple[RandomVariable]
+):
+    """Build a NUTS kernel.
+    """
+
+    # Work in the transformed space ----------------------------------
```

Comment:
Let's not commit any custom formatting&mdash;comments or otherwise.  We need to keep the code style in the `aesara-devs` projects as simple and uniform as possible (and reasonable).

Reply:
Yes, these are temporary. And probably a sign that my workflow can be perfected.